File: Big_JS_Algos/Misc_DS_DEV/Leetcode/Frequency_Counter/ransomNote.js
//Frequency Counter:

const canConstruct = (ransomNote, magazine) => {
  if (magazine.length === 0) return false;

  let lookup = {};

  for (let i = 0; i < magazine.length; i++) {
    lookup[magazine[i]]
      ? (lookup[magazine[i]] += 1)
      : (lookup[magazine[i]] = 1);
  }

  for (let i = 0; i < ransomNote.length; i++) {
    if (!lookup[ransomNote[i]]) {
      return false;
    } else {
      lookup[ransomNote[i]] -= 1;
    }
  }

  return true;
};

File: Big_JS_Algos/Technical_Interview_Prep/Leetcode/Frequency_Counter/ransom_note.py

def can_construct(ransom_note, magazine):
    if len(magazine) == 0:
        return False

    lookup = {}

    for char in magazine:
        if char in lookup:
            lookup[char] += 1
        else:
            lookup[char] = 1

    for char in ransom_note:
        if char not in lookup or lookup[char] == 0:
            return False
        else:
            lookup[char] -= 1

    return True